locust_benchmark/README.md

```diff
@@ -0,0 +1,178 @@
+
+This section illustrates how to quickly run locust in a single process, see the next section on how to scale locust with a master and multiple worker processes.
+
+#### Prerequisites
+
+1. Hopsworks cluster
+   1. You can follow the documentation on how to setup a cluster on [managed.hopsworks.ai](https://docs.hopsworks.ai/3.1/setup_installation/aws/cluster_creation/)
+   2. Ideally you create an extra node in the same VPC as the Hopsworks cluster, which you can use to run the benchmark suite from. The easiest way to do this is by using a dedicated [API node](https://docs.hopsworks.ai/3.1/setup_installation/common/rondb/#api-nodes).
+   3. Once the cluster is created you need the following information:
+      - Hopsworks domain name: **[UUID].cloud.hopsworks.ai** which you can find in the *Details* tab of your managed.hopsworks.ai control plane
+      - A project and its name on the hopsworks cluster. For that log into your cluster and create a project.
+      - An API key for authentication. Head to your project and [create an API key for your account](https://docs.hopsworks.ai/3.1/user_guides/projects/api_key/create_api_key/).
+2. Python environment
+   In order to run the benchmark suite, you need a Python environment.
+   For that we will SSH into the created API node. Make sure that [SSH is enabled](https://docs.hopsworks.ai/3.1/setup_installation/common/services/) on your cluster.
+
+   1. On AWS all nodes have public IPs, so you can head to your cloud console to find the IP of the API node and SSH directly:
+      `ssh ubuntu@[ip-of-api-node] -i your-ssh-key.pem`
+      On Azure, you will have to proxy through the Hopsworks head node, you can either use the public IP or the Hopsworks domain name as noted above.
+      On GCP, we don't attach SSH keys, but you can use the GCP CLI to create an ssh session.
+
+   2. API nodes come with Anaconda pre-installed, which we can use to create a python environment and install our dependencies:
+
+      ```bash
+      conda init bash
+      ```
+
+      After this you need to create a new SSH session.
+
+      ```bash
+      conda create -n locust-benchmark python=3.9
+      conda activate locust-benchmark
+      ```
+
+      Clone the feature-store-api repository to retrieve the benchmark suite and install the requirements.
+      Make sure to checkout the release branch matching the Hopsworks version of your cluster.
+
+      ```bash
+      git clone https://github.com/logicalclocks/feature-store-api.git
+      cd feature-store-api
+      git checkout master
+      cd locust_benchmark
+      pip install -r requirements.txt
+      ```
+
+#### Creating feature group for lookups
+
+1. Save the previously created API key in a file named `.api_key`
+
+```bash
+echo "[YOUR KEY]" > .api_key
```

Comment:
Perhaps just add an env file, with all configs, which you source? Or just write it into the config?